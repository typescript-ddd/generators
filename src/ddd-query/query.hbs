import { Query } from "@typescript-ddd/core";

/**
 * Represents a query for finding {{noun entityName}}.
 {{#each properties}}
 * @param {{lbrace}}{{type}}{{rbrace}} {{name}} - The {{natural name}} of the {{natural ../entityName}}.
  {{/each}}
 */
export class {{actionName}}{{pascal entityName}}Query implements Query {
  constructor(
    {{#each properties}}
    public readonly {{name}}: {{type}},
    {{/each}}
  ) {
    // TODO: Add validation here
  }
}
