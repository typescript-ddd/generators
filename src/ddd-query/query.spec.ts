import { QueryTemplateData } from "./query.types";
import { generateQuery } from "./query.generator";

describe("Query", () => {

  it("should render template", () => {
    // Arrange
    const data: QueryTemplateData = {
      entityName: "User",
      properties: [
        { name: "id", type: "UserId" },
      ],
    };

    // Act
    const result = generateQuery(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
