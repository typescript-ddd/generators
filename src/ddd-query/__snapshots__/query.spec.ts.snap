// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query should render template 1`] = `
"import { Query } from "@typescript-ddd/core";

/**
 * Represents a query for finding a user.
 * @param {UserId} id - The id of the user.
 */
export class FindUserQuery implements Query {
  constructor(
    public readonly id: UserId,
  ) {
    // TODO: Add validation here
  }
}
"
`;
